src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -696,22 +699,32 @@ private double calculateSuccessfulTestPoints(final ProgrammingExercise programmi
             return credits;
         }).sum();
 
-        /*
-         * The points are capped by the maximum achievable points. The cap is applied before the static code analysis penalty is subtracted as otherwise the penalty won't have any
-         * effect in some cases. For example with maxPoints=20, successfulTestPoints=30 and penalty=10, a student would still receive the full 20 points, if the points are not
-         * capped before the penalty is subtracted. With the implemented order in place successfulTestPoints will be capped to 20 points first, then the penalty is subtracted
-         * resulting in 10 points.
-         */
+        return capPointsAtMaximum(programmingExercise, successfulTestPoints);
+    }
+
+    /**
+     * Caps the points at the maximum achievable number.
+     *
+     * The cap should be applied before the static code analysis penalty is subtracted as otherwise the penalty won't have any effect in some cases.
+     * For example with maxPoints=20, points=30 and penalty=10, a student would still receive the full 20 points, if the points are not
+     * capped before the penalty is subtracted. With the implemented order in place points will be capped to 20 points first, then the penalty is subtracted
+     * resulting in 10 points.
+     *
+     * @param programmingExercise Used to determine the maximum allowed number of points.
+     * @param points A number of points that may potentially be higher than allowed.
+     * @return The number of points, but no more than the exercise allows for.
+     */
+    private double capPointsAtMaximum(final ProgrammingExercise programmingExercise, double points) {
         double maxPoints = programmingExercise.getMaxPoints() + Optional.ofNullable(programmingExercise.getBonusPoints()).orElse(0.0);
 
-        if (Double.isNaN(successfulTestPoints)) {
-            successfulTestPoints = 0;
+        if (Double.isNaN(points)) {
+            points = 0;
```

Comment:
Consider not reassigning parameters as suggested by codacy

Reply:
https://github.com/ls1intum/Artemis/tree/refactor/programming/grading-service

I actually fixed that in a 19th commit yesterday, but Github for whatever reason only shows it in the branch but not in here.